src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -238,7 +238,7 @@ public GlobalSearchBar(LibraryTabContainer tabContainer,
     }
 
     private void initSearchModifierButtons() {
-        fulltextButton.setSelected(searchPreferences.isFulltext());
+        fulltextButton.setSelected(searchPreferences.isFulltext() && filePreferences.shouldFulltextIndexLinkedFiles());
```

Comment:
We cannot do some binding here, I assume.

But some subscription on ´filePreferences.shouldFulltextIndexLinkedFiles()`? If that value changes to `false`, the button should be disabled. - I think, it is also OK to subscribe to `true` and to enable the button. -- OK, we also need to update `isFullText` accordingly?